backend/api/Controllers/MissionController.cs

```diff
@@ -384,31 +385,49 @@ public async Task<ActionResult<byte[]>> GetMap([FromRoute] string installationCo
 
         if (area == null)
         {
-            // This is disabled for now as the Area database is not yet populated
+            // This is disabled for now as the Area database is not yet populated. We may want to auto populate it 
+            // by using the location info in the echo tags.
             //return NotFound($"Could not find area with name {scheduledMissionQuery.AreaName} in installation {scheduledMissionQuery.InstallationCode}");
         }
 
-        // TODO: search for if a source with the given type and URL exists, then reuse it
+        var source = await _sourceService.ReadBySourceIdAndType($"{echoMission.Id}", MissionSourceType.Echo);
+        MissionDefinition? missionDefinition = null;
 
-        var scheduledMissionDefinition = new MissionDefinition
+        if (source == null)
         {
-            Id = Guid.NewGuid().ToString(),
-            Source = new Source
+            source = new Source
             {
                 SourceId = $"{echoMission.Id}",
                 Type = MissionSourceType.Echo
-            },
-            Name = echoMission.Name,
-            InspectionFrequency = scheduledMissionQuery.InspectionFrequency,
-            InstallationCode = scheduledMissionQuery.InstallationCode,
-            Area = area
-        };
+            };
+        }
+        else
+        {
+            var missionDefinitions = await _missionDefinitionService.ReadMissionDefinitionsBySourceId(source.SourceId);
+            // In theory there could be more than one mission definition that is relevant, but we need to randomly choose one
+            missionDefinition = missionDefinitions.FindAll(m => area == null || (m.Area != null && m.Area.Id == area.Id))[0];
```

Comment:
In what scenario would multiple missionDefinitions map to the same echo mission?
If that is the case, randomly choosing one sounds dangerous eheh

Reply:
Currently we don't imagine any situation where this would happen, since each echo mission is tied to a specific area and so it doesn't make sense to have more than one definition, and the only difference they could have would be comment and inspection frequency. The only way for this to happen would be through manual database editing. As a result it shouldn't be dangerous to choose one at random, especially as this would update the lastRun for the given missionDefinition, preventing confusion where a mission has recently been run but we have not detected an inspection in the required inspection frequency time window. The only way we could prevent the need to assume more than one mission definition per source would be if a mission definition owns a source in the database but that is not currently the case. But even in this case we would still use the same code in the mission controller.

Reply:
If we are worried about it though we could just create a new plain mission definition if there are currently none that don't have the optional values not set.

Reply:
Currently I don't think there is anything tying an echo mission to a single area, other than we can choose to do so. With the ability to traverse between areas I think we need to assume that it is likely that an echo mission may span many areas.

Reply:
I suppose in this case, we would then only have an issue if more than one mission definition used the same source, whilst also mapping to different areas. I think we can likely safely have a mission definition own a source, since I don't see a scenario where we would need more than one mission definition for the same tasks. In that case we could also just create a new Source object. To simplify the database we can then make a mission definition own a source, but I think this may be outside the scope of this PR. If you disagree though then I can try add this.